packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -177,17 +182,59 @@ export class Toolkit extends CloudAssemblySourceBuilder implements AsyncDisposab
             usePreviousParameters: parameters?.keepExistingParameters,
           },
         );
+
+        const result: EnvironmentBootstrapResult = {
+          environment: environment.name,
+          status: bootstrapResult.noOp ? 'no-op' : 'success',
+          message: bootstrapResult.noOp ? 'No changes required' : 'Successfully bootstrapped',
+          duration: Date.now() - envStartTime,
+        };
+
+        results.push(result);
+
         const message = bootstrapResult.noOp
           ? ` ✅  ${environment.name} (no changes)`
           : ` ✅  ${environment.name}`;
 
         await ioHelper.notify(IO.CDK_TOOLKIT_I9900.msg(chalk.green('\n' + message), { environment }));
         await bootstrapSpan.end();
       } catch (e: any) {
+        const result: EnvironmentBootstrapResult = {
+          environment: environment.name,
+          status: 'failed',
+          error: e,
+          message: formatErrorMessage(e),
+          duration: Date.now() - envStartTime,
+        };
```

Comment:
we throw an exception three lines  further down, so this will just never happen.

Reply:
Got it, removed this code. I also removed 'failed' as a status case because it throws an error.